document/core/exec/runtime.rst

```diff
@@ -277,24 +315,26 @@ It is an invariant of the semantics that the length of the element vector never
 Memory Instances
 ~~~~~~~~~~~~~~~~
 
+.. todo:: We used to update the memory type when a memory is grown. This does not work for shared memories. In fact, the "current" type of a shared memory is nondeterministic. We need to model that during instantiation somehow.
+
 A *memory instance* is the runtime representation of a linear :ref:`memory <syntax-mem>`.
-A memory instance records an optional maximum size for a memory, if one was specified at the definition site of the memory, and a flag indicating whether the memory can be :ref:`shared <syntax-shared>` across threads.
+It records its original :ref:`memory type <syntax-memtype>`
+and takes one of two different forms depending on whether that type is :ref:`shared <syntax-shared>` or not.
 
 .. math::
    \begin{array}{llll}
    \production{(memory instance)} & \meminst &::=&
-     \{ \MIMAX~\u32^?, \MISHARE~\UNSHARED, \MIDATA~\vec(\byte) \} \\ &&|&
-     \{ \MIMAX~\u32^?, \MISHARE~\SHARED \} \\
+     \{ \MITYPE~\memtype, \MIDATA~\vec(\byte) \} \\ &&|&
```

Comment:
syntactically there's nothing here stopping memtype from being `shared`, even if there's a byte vector, right?

Reply:
Yes, it's just an invariant of the semantics (I actually had a statement about that initially, maybe I should reinstate it for clarity).